Adding GRE tunnel support for ubuntu/debian

sepulworld opened the pull request:
Provide the ability to create a GRE tunnel between to hosts.

adrienthebo commented:
Sorry for the delay in responding to this, I haven't been able to prioritize work on this module. It looks like the parameters added in this pull request are just generic interface options. The Debian `network_config` provider can accept those as a hash of key/value pairs so these don't have to be native type values, they can be handled like the bond defines (https://github.com/puppet-community/puppet-network/blob/master/manifests/bond/debian.pp#L27).

sepulworld commented:
Nice ok, I'll update. I still need to work through red hat distros.

igalic commented:
heh, @adrienthebo that reduces my chances to ever get #83 in.

adrienthebo commented:
@igalic how so?

igalic commented:
maybe it just means i need to look into how rhel does it.
either that, or we need to speed up that change in puppet types&providers where we allow direct access to the provider from within  manifests ;)

adrienthebo commented:
You already can pass arbitrary options to the provider through the options hash, so you need more than that?

igalic commented:
nothing more than a regular reminder that i can do that ~_~

sepulworld commented:
@adrienthebo I rebased and reworked the pull request to use defines, leveraging the hash options.  Going to add a gre/redhat.pp and will test it out.  Stand by...

igalic commented on manifests/gre.pp:
```diff
@@ -0,0 +1,128 @@
+
+        mode             => $mode,
+        ttl              => $ttl,
+
```
is there a special reason why this is weirdly indented?

sepulworld replied:
Ah, good question.  I just fixed the indents.  Thanks @igalic

adrienthebo commented:
@sepulworld thanks! @igalic yay? nay?

igalic commented:
@sepulworld can you [squash](http://www.git-scm.com/book/en/Git-Tools-Rewriting-History#Squashing-Commits) this down to one or two commits, please?

sepulworld commented:
I can... Ill so that tonight

On Wed, Mar 11, 2015, 6:15 AM Igor Galić <notifications@github.com> wrote:

> @sepulworld <https://github.com/sepulworld> can you squash
> <http://www.git-scm.com/book/en/Git-Tools-Rewriting-History#Squashing-Commits>
> this down to one or two commits, please?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppet-community/puppet-network/pull/55#issuecomment-78260207>
> .
>

sepulworld commented:
Apparently I can't squash and recommit?

https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

I can submit a new pull request with a new squashed branch.  Would that work?

sepulworld commented:
Sorry, failed rebase squash attempt.  I opened up clean pull request #97

ffrank commented:
On 03/26/2015 07:01 AM, Zane wrote:
> Apparently I can't squash and recommit?
> 
> https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

Hm, those people are either ignorant or very stuck in their ways.

For the record, pushing a rebased branch to github is no problem as long
as you use `git push -f`. This is the encouraged way of working with
PRs, because they *will* get updated with your new commits.
